apps/dapp/src/components/Pages/Ascend/components/LBPForm/use-pool-contract.ts

```diff
@@ -0,0 +1,94 @@
+import { Contract } from 'ethers';
+
+import balancerPoolAbi from 'data/abis/balancerPool.json';
+import { Pool } from 'components/Layouts/Ascend/types';
+import { useWallet } from 'providers/WalletProvider';
+import { useState } from 'react';
+import { DecimalBigNumber } from 'utils/DecimalBigNumber';
+
+export const usePoolContract = (pool?: Pool) => {
+  const { signer } = useWallet();
+
+  if (!pool || !signer) {
+    return {
+      setSwapEnabled: {
+        handler: async () => {
+          throw new Error('Wallet not connected');
+        },
+        isLoading: false,
+        error: null,
+      },
+      updateWeightsGradually: {
+        handler: async () => {
+          throw new Error('Wallet not connected');
+        },
+        isLoading: false,
+        error: null,
+      },
+    };
+  }
+
+  const [isSetSwapEnabledLoading, setIsSetSwapEnabledLoading] = useState(false);
+  const [setSwapEnabledError, setSetSwapEnabledError] = useState<null | string>(null);
+
+  const [isUpdateWeightsGraduallyLoading, setIsUpdateWeightsGraduallyLoading] = useState(false);
+  const [updateWeightsGraduallyError, setUpdateWeightsGraduallyError] = useState<null | string>(null);
+
+  const poolContract: Contract = new Contract(pool.address, balancerPoolAbi, signer);
+
+  const setSwapEnabledHandler = async (enabled: boolean) => {
+    setIsSetSwapEnabledLoading(true);
+    setSetSwapEnabledError(null);
+    try {
+      const result = await poolContract!.setSwapEnabled(enabled, {
+        gasLimit: 400000,
+      });
+      await result.wait();
+    } catch (error: any) {
+      console.error(error);
+      setSetSwapEnabledError('Transaction failed.');
```

Comment:
if it made sense (not sure if it does) could use the NotificationManager context and just show notifications on create/error.

Reply:
https://github.com/TempleDAO/temple/pull/461/commits/bb507608688ba733e68f2319522ec1215bc9784f